### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by SimonR99**
Add navigation, 2D slam, 3D slam (RTABMAP), frontier exploration, IMU/velodyne support

### CODE REVIEW

File: markhor_slam/launch/markhor_slam_2d.launch [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    <group ns="markhor">
+        <!-- RobotLocalization (Sensor fusion) and tf -->
+        <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization"
+            clear_params="true">
+            <rosparam command="load" file="$(find markhor_slam)/config/ekf_localization.yaml" />
+        </node>
+    </group>
+
+
```
Comment (Leuchak) [inline markhor_slam/launch/markhor_slam_2d.launch:28]: trailing whitespace

**Review by Leuchak — CHANGES_REQUESTED** [review]
Good work

File: markhor_navigation/scripts/explore_and_return.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    goal.target_pose.pose = return_position
+    client.send_goal(goal)
+
+    # Wait for the action to complete
+    client.wait_for_result()
+
+    return StartExplorationResponse(True, "Exploration started and return position set successfully")
+
+if  __name__ == '__main__':
+    try:
```
Comment (Leuchak) [inline markhor_navigation/scripts/explore_and_return.py:58]: oops, I meant 2 \n

**Review by Leuchak — CHANGES_REQUESTED** [review]

**Review by Leuchak — APPROVED** [review]
look good

File: markhor_gazebo/launch/local.launch
Comment (saxtot) [inline markhor_gazebo/launch/local.launch:1]: Old file, to be removed.

File: markhor_navigation/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  std_msgs
+  geometry_msgs
+)
+
+catkin_package(
+
+    # INCLUDE_DIRS include
+    # LIBRARIES markhor_navigation
+    # CATKIN_DEPENDS move_base my_odom_configuration_dep my_sensor_configuration_dep my_tf_configuration_dep
+    # DEPENDS system_lib
```
Comment (saxtot) [inline markhor_navigation/CMakeLists.txt:28]: Unnecessary comments. Same thing on lines 34 and 35.

File: markhor_navigation/launch/markhor_configuration.launch
Comment (saxtot) [inline markhor_navigation/launch/markhor_configuration.launch:1]: Empty file, to be removed.

File: markhor_navigation/scripts/explore_and_return.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+if __name__ == '__main__':
+    try:
+        # Initialize the ros node
+        rospy.init_node('explore_and_return', anonymous=True)
+
+        # Start the service server
+        rospy.Service('start_exploration', StartExploration, handle_start_exploration)
+
+        # Keep the node running until it's shut down
+        rospy.spin()
```
Comment (saxtot) [inline markhor_navigation/scripts/explore_and_return.py:67]: Might want to consider a ROS [rate](https://roboticsbackend.com/ros-rate-roscpy-roscpp/).

File: markhor_slam/launch/markhor_slam_2d.launch [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    <node name="pointcloud_to_laserscan_node" pkg="pointcloud_to_laserscan"
+        type="pointcloud_to_laserscan_node" respawn="false" output="screen">
+        <remap from="/cloud_in" to="/velodyne_points" />
+        <param name="min_height" value="-0.3" />
+        <param name="max_height" value="0.3" />
+        <param name="angle_min" value="-3.14" />
+        <param name="angle_max" value="3.14" />
+        <param name="range_min" value="1.2" />
+        <param name="range_max" value="10" />
+        <param name="use_inf" value="false" />
```
Comment (saxtot) [inline markhor_slam/launch/markhor_slam_2d.launch:17]: Would be cool to have the units of measurement associated as comments.

File: markhor_slam/launch/markhor_slam_2d.launch [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    <arg name="lidar_args"
+        default="
+      --Reg/Strategy 1 
+      --RGBD/NeighborLinkRefining true 
+      --Grid/CellSize $(arg cell_size) 
+      --Icp/PointToPlaneRadius 0 
+      --Icp/MaxTranslation 1" />
+
+    <!---
+    Run rtabmap -->
```
Comment (saxtot) [inline markhor_slam/launch/markhor_slam_2d.launch:37]: yuck

**Review by saxtot — COMMENTED** [review]
Some things to change. Great work overall.

**Pull request merged**